### PR TITLE
LPS-49419 Security bugs in Pingback

### PR DESCRIPTION
**Pull request by topolik**
Hi Sam,

can you please peer review? 

https://issues.liferay.com/browse/LPS-49419

Thanks!

### CODE REVIEW

File: modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/internal/util/PingbackMethodImpl.java [hunk 1]
```diff
@@ -388,6 +394,30 @@ protected Response validateSource() throws Exception {
...
 
+	private boolean _isSourceURILocalNetwork() {
+		try {
+			URI uri = new URI(_sourceURI);
+
+			InetAddress ipaddress = InetAddress.getByName(uri.getHost());
+
+			if (ipaddress.isLoopbackAddress() ||
+				ipaddress.isLinkLocalAddress() ||
+				ipaddress.isSiteLocalAddress()) {
```
Comment (samuelkong) [inline modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/internal/util/PingbackMethodImpl.java]: @topolik 
A little beyond my networking knowledge here. Do we also need `isAnyLocalAddress()` to check for wildcard addresses?

Comment (topolik) [reply modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/internal/util/PingbackMethodImpl.java]: That should be fine. Looks like `isAnyLocalAddress()` checks `0.0.0.0` that is used to listen on all interfaces. Nothing that could be a target IP to connect to.

**Comment by samuelkong** [comment]
Thanks. Forwarded to Brian, https://github.com/brianchandotcom/liferay-portal/pull/60869

**Comment by topolik** [comment]
@samuelkong , you were correct, I just tried it and it's possible to target `0.0.0.0`!

**Comment by topolik** [comment]
I'll fix it and send to Brian, thanks!

**Comment by samuelkong** [comment]
@topolik FYI: Looks like we also need to fix a test failure. https://github.com/brianchandotcom/liferay-portal/pull/60869#issuecomment-400990691

**Comment by topolik** [comment]
Thanks, yes, I forgot.

**Comment by topolik** [comment]
Please see https://github.com/brianchandotcom/liferay-portal/pull/60885. Thanks.

**Pull request closed without merging**